editor/js/results.js

```diff
@@ -283,6 +284,9 @@ class ResultsPresenter {
     const years = self._results.getYears(self._filterSet.getWithYear(null));
     self._filterSet = self._filterSet.getWithYear(Math.max(...years));
 
+    // Update dimension labels based on current filter settings
+    self._dimensionManager.updateSubstancesLabel(self._filterSet);
```

Comment:
Let's push this down into a method called `updateSubstancesLabel` that lives inside `DimensionPresenter` to avoid referring to private (leading underscore) members.